### PR TITLE
Mock locations

### PR DESCRIPTION
**Pull request by oliv3**
Detect if a location is a mock

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -2,4 +2,4 @@ distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-4.6-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-4.5.1-all.zip
```
Comment (tananaev) [inline gradle/wrapper/gradle-wrapper.properties]: Why are you downgrading gradle?

File: app/src/main/java/org/traccar/client/Position.java [hunk 2]
```diff
@@ -139,4 +140,7 @@ public void setBattery(double battery) {
         this.battery = battery;
     }
 
+    private boolean mock;
+
+    public boolean getMock() { return mock; }
```
Comment (tananaev) [inline app/src/main/java/org/traccar/client/Position.java]: Setter is missing.

File: app/src/main/java/org/traccar/client/ProtocolFormatter.java [hunk 3]
```diff
@@ -34,7 +34,8 @@ public static String formatRequest(String url, Position position, String alarm)
                 .appendQueryParameter("bearing", String.valueOf(position.getCourse()))
                 .appendQueryParameter("altitude", String.valueOf(position.getAltitude()))
                 .appendQueryParameter("accuracy", String.valueOf(position.getAccuracy()))
-                .appendQueryParameter("batt", String.valueOf(position.getBattery()));
+                .appendQueryParameter("batt", String.valueOf(position.getBattery()))
+                .appendQueryParameter("mock", String.valueOf(position.getMock()));
```
Comment (tananaev) [inline app/src/main/java/org/traccar/client/ProtocolFormatter.java]: Make sure all unit tests are still working. Ideally update unit tests as well.

File: app/src/main/java/org/traccar/client/DatabaseHelper.java [hunk 4]
```diff
@@ -100,6 +101,7 @@ public void insertPosition(Position position) {
         values.put("course", position.getCourse());
         values.put("accuracy", position.getAccuracy());
         values.put("battery", position.getBattery());
+        values.put("mock", position.getMock() ? 1 : 0);
```
Comment (tananaev) [inline app/src/main/java/org/traccar/client/DatabaseHelper.java]: Why is `select` not updated? I would recommend to include check in unit tests.

File: app/build.gradle [hunk 5]
```diff
@@ -7,7 +7,7 @@ android {
     defaultConfig {
         applicationId "org.traccar.client"
         buildConfigField "boolean", "HIDDEN_APP", "false"
-        minSdkVersion 15
+        minSdkVersion 18
```
Comment (tananaev) [inline app/build.gradle]: Please leave min SDK as it was.

File: app/src/main/java/org/traccar/client/Position.java (hunk 2 above)
Comment (oliv3) [reply app/src/main/java/org/traccar/client/Position.java]: Done

Comment (tananaev) [reply app/src/main/java/org/traccar/client/Position.java]: It's still not fixed.

Comment (oliv3) [reply app/src/main/java/org/traccar/client/Position.java]: Which line please ?

Comment (tananaev) [reply app/src/main/java/org/traccar/client/Position.java]: You need a line break after open bracket and before closing bracket.

Comment (oliv3) [reply app/src/main/java/org/traccar/client/Position.java]: Well I tried to stick to the existing style, hope it's ok now

**Comment by tananaev** [comment]
I have tested unit tests on your branch and they work just fine. Are you sure you selected "regular" flavor and not "google" one?

**Comment by oliv3** [comment]
I just left the flavor to the default, which seems to be Google.

Had to wander through those thousands of menus to finally switch to "regular", and now everything is OK :)

I think documentation needs some update

Thanks !

**Comment by tananaev** [comment]
Sorry, but obviously you have to keep original prefixes. In future I would recommend to not change something that you don't fully understand.

**Comment by oliv3** [comment]
In future I would recommend helping people understanding things they don't fully understand instead of bashing. You certainly have your code published, but as for bringing people to contribute, that's a Zero.

Great work, though.

File: app/build.gradle [hunk 6]
```diff
@@ -29,12 +29,12 @@ android {
...
-    compile 'com.mapzen.android:lost:3.0.4'
-    testCompile 'junit:junit:4.12'
-    testCompile 'org.robolectric:robolectric:3.6.1'
-    googleCompile 'com.google.firebase:firebase-core:11.8.0'
-    googleCompile 'com.crashlytics.sdk.android:crashlytics:2.9.1'
+    implementation 'com.android.support:design:27.1.1'
+    implementation 'com.mapzen.android:lost:3.0.4'
+    testImplementation 'junit:junit:4.12'
+    testImplementation 'org.robolectric:robolectric:3.6.1'
+    googleApi 'com.google.firebase:firebase-core:15.0.0'
```
Comment (tananaev) [inline app/build.gradle]: This is still not fixed.

Comment (oliv3) [reply app/build.gradle]: Any hint on what you want would be useful

Comment (tananaev) [reply app/build.gradle]: Are you serious? It should be implementation not api.

Comment (oliv3) [reply app/build.gradle]: Would be quicker if you yell me what you want in there.

Comment (tananaev) [reply app/build.gradle]: It should be `googleImplementation` not `googleApi`.

**Comment by tananaev** [comment]
One more thing I just though about. I think we should send `mock` parameter only if it's true.

**Comment by tananaev** [comment]
Merged, thanks.

**Pull request merged**